### DISCUSSION TITLE
spectrum information for each link

### DISCUSSION THREAD

**Discussion by akhil-mutike** [Telecominfraproject/oopt-gnpy#491] [Q&A] [answered]
Hello community,
Presently i am working low margin optical design.To obtain exact OSNR of particular light path in optical network  involves considering the effect of other other active lightpaths in the Network.I am quite new to GNpy , I have few questions regarding the spectral loading in GNPY.
1) can we have seperate spectral loading for each link? 
2) can we have non uniform spectral load in a single link(each channel with different mod order,baudrate..etc)?

**Comment by jktjkt** [comment, marked as answer]
> 1. can we have seperate spectral loading for each link?

The core part of GNPy supports this use case just fine, but our "frontend" code that you can invoke form the CLI, working with JSON or XLS files as inputs, assumes the same spectrum loading over the whole network.

> 2. can we have non uniform spectral load in a single link(each channel with different mod order,baudrate..etc)?

[Yes.](https://gnpy.readthedocs.io/en/master/json.html#arbitrary-channel-definition)

**Reply by akhil-mutike** [reply]
thanks  ,can you please point me to the resources ,which can help me in accessing/understanding the core part of GNPy(i assume these are bunch of pyhton functions/classes).like detailed documentation regarding the GNPy code itslef.

**Reply by jktjkt** [reply]
Yes, the "core part" is the Python code of GNPy itself. The documentation is in the code, and we do not have any additional resources besides the inline comments and docstrings.

**Reply by akhil-mutike** [reply]
thank you , i went through the elements.py file. as far i could undestand, each element (may be fiber,ROADM,edfa..) takes in parameters and spectral information and propogates the the signal  and gives out a modified spectral information.so, i should 'propogate' the signal through each element with corresponding spectral information(while calcualting the change in specral load  at each node(adding or droping frequency) and coreespondingly changing the spectral information input for each span)  .seems like humungous task for a network like CORONET.Any suggestions or tricks to making  it easier to implement.

**Reply by akhil-mutike** [reply]
hi @jktjkt ,
could you please help me in further queries?
I have one morequestion to add.
while passing spectralinfo to an element, i need to switchoff a certain channel.is it achieved by simply skipping the frequency in the frequency array?
Thanks ,
Akhil
